File: week1/mergeSort.py
import os
import sys


def mergeSort( array, splitInv=0):
	arrayLength = len(array)
	if arrayLength == 1:
		return array, splitInv

	lSort,lInv = mergeSort( array[:arrayLength/2], splitInv=0 )
	rSort,rInv = mergeSort( array[arrayLength/2:], splitInv=0 )

	splitInv += lInv + rInv
	li=ri=ai=0

	while li < len( lSort):
		if ri < len( rSort):
			if lSort[li] <= rSort[ri]:
				array[ai] = lSort[li]
				li += 1
			else:
				array[ai] = rSort[ri]
				ri += 1
				splitInv += len( lSort) - li
		else:
			array[ai] = lSort[li]
			li += 1
		ai += 1

	while ri < len( rSort):
		array[ai] = rSort[ri]
		ri += 1
		ai += 1

	return array, splitInv


if __name__ == "__main__":
	f = open( sys.argv[1], 'r')
	toBeSorted = f.readlines()
	toBeSorted = [int(l.strip()) for l in toBeSorted if l != ""]
	print mergeSort( toBeSorted)
	#print mergeSort( [1,2,4,3,5,6])
	#print mergeSort( [1,3,5,2,4,6])

File: week1/mergeSort.cpp
#include <iostream>
#include <vector>
using namespace std;


void printArray( vector<int> * array ) {
	int l = (*array).size();
	for (int i=0; i < l; i++) {
		cout << (*array)[i] << ", ";
	}
	cout << endl;
}

vector<int> *  mergeSort( vector<int> * array, vector<int> * outArray, int * splitInv ) {
	int arrayLength = (*array).size();
	int lSize = arrayLength/2;

	vector<int> lSort( lSize);
	vector<int> rSort( arrayLength - lSize);

	if ( arrayLength == 1 ) {
		return array;
	}

	
}



int main() {
	vector<int> array1;
	for (int i=5; i >= 0; i--) {
		array1.push_back(i);
	}
	printArray( &array1);

	//mergeSort( array1 );

	printArray( &array1);
	
	vector<int> a(2);
	vector<int> b;

	b.push_back(5);
	b.push_back(2);

	a = b;
	a[0] = 33;
	printArray( &a);
	printArray( &b);

}


